Add date and datetime input enhancement (#276)
AY2324S2-CS2103-F15-1/tp: docs/DeveloperGuide.md

_{Explain here how the data archiving feature will be implemented}_

--------------------------------------------------------------------------------------------------------------------

## **Planned Enhancements**

This section describes planned features to be implemented after v1.4.

### Flexibility of Date and DateTime Inputs

In the current implementation of FINDvisor, it only strictly accepts dates of the format `dd-MM-yyyy` and datetime values of the format `dd-MM-yyyy`T`HH:mm`.
To enhance usability, FINDvisor will accept the following formats:

* Date:
  * `d-M-yyyy` - Accepts single digit days and months without adding any zeroes to the front. Double-digit days and months will still be accepted in this format.

>   * `d-M-yy` - Similar to the format above, except that it allows 2 digit values for the year as input. When using a double-digit value for year, it will use the current century to produce the full date e.g., `1-1-24` will be read as the `1st January 2024`.

Reviewer: I feel like its not necessary to give date specifications since our current implementation of meeting only use dateTime.

Author: I think date is being used for find command?

Reviewer: Alright fair point.

- Modify the pre-defined set of allowed characters to **capture any character instead**. This allows for flexibility since all special characters will be accepted for the `ADDRESS` field.
- For addresses containing `/` characters, the user will be required to use the `\` character to escape each `/` character similar to the example in Planned Enhancement 1.

> ### 3. Allow single day, month, hour and minute as valid Date and DateTime inputs

Reviewer 2: ```suggestion
### 3. Increase flexibility of Date and DateTime formats
```

**Current Implementation**:

- FINDvisor only strictly accepts `DATE` of the format `dd-MM-yyyy` and `DATETIME` of the format `dd-MM-yyyy`T`HH:mm`.
- This requires single digit day and month values to be padded with a zero to be accepted by FINDvisor, hindering the ease of use of the function.

> - For example, if the user wishes to schedule a meeting to a `START_DATETIME` of `1/1/2024T1:1`, which is equivalent to a valid datetime `01/01/2024T01:01`, FINDvisor will recognize the `START_DATETIME` value as invalid as it does not comply to our specified format.

Author: The example is meant to show that minutes would be accepted with single digit value as well. Do you think changing it to a single 0 would be better?
